extension/src/completion/componentCompletionItemProvider.ts

```diff
@@ -0,0 +1,56 @@
+import { IdCompletionProvider } from "./IdCompletionProvider";
+import * as vscode from "vscode";
+export class ComponentCompletionItemProvider implements vscode.CompletionItemProvider {
+
+    private completionProviders = [
+        { tag: "StateData", attribute: "Id", provider: new IdCompletionProvider("StateData") }
+    ];
+
+    public provideCompletionItems(
+        document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken):
+        Thenable<vscode.CompletionItem[]> {
+        return new Promise<vscode.CompletionItem[]>((resolve, reject) => {
+            const documentTextRange = new vscode.Range(new vscode.Position(position.line, 0), position);
+            const textBeforeCursor = document.getText(documentTextRange);
```

Comment:
so it only works if tag and attributes are on the same line, right?

Reply:
Yes I know this limitation. I have an idea for this fix but will do it later on